statistics: add some doc for the exp feature (#9891)
pingcap/docs: system-variables.md

    - `RESTRICTED_VARIABLES_ADMIN`: The ability to see and set sensitive variables in `SHOW [GLOBAL] VARIABLES` and `SET`.
    - `RESTRICTED_USER_ADMIN`: The ability to prevent other users from making changes or dropping a user account.

### tidb_enable_extended_stats

> - Scope: SESSION | GLOBAL

Reviewer: ```suggestion
- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
```

- Type: Boolean
- Default value: `OFF`

> - This variable indicates whether TiDB can collect the extended statistic to guide the optimizer. Refer to the chapter [Introduction to Extended Statistics](/extended-statistics.md) for more information.

Reviewer: ```suggestion
- This variable indicates whether TiDB can collect the extended statistic to guide the optimizer. See [Introduction to Extended Statistics](/extended-statistics.md) for more information.
```

> - This variable indicates whether TiDB can collect the extended statistic to guide the optimizer. Refer to the chapter [Introduction to Extended Statistics](./extended-statistics.md) for more information.

Reviewer 2: ```suggestion
- This variable indicates whether TiDB can collect the extended statistic to guide the optimizer. Refer to the chapter [Introduction to Extended Statistics](/extended-statistics.md) for more information.
```
